src/AWS.Deploy.Orchestration/OptionSettingHandler.cs

```diff
@@ -333,5 +349,79 @@ public bool IsSummaryDisplayable(Recommendation recommendation, OptionSettingIte
 
             return true;
         }
+
+        /// <summary>
+        /// Checks whether the option setting item has been modified by the user. If it has been modified, then it will hold a non-default value
+        /// </summary>
+        /// <returns>true if the option setting item has been modified or false otherwise</returns>
+        public bool IsOptionSettingModified(Recommendation recommendation, OptionSettingItem optionSetting)
+        {
+            // If the option setting is not displayable, that means its dependencies are not satisfied and it does not play any role in the deployment.
+            // We do not need to evaluate whether it has been modified or not.
+            if (!IsOptionSettingDisplayable(recommendation, optionSetting))
+            {
+                return false;
+            }
+
+            if (optionSetting.Type.Equals(OptionSettingValueType.List))
+            {
+                var currentSet = GetOptionSettingValue<SortedSet<string>>(recommendation, optionSetting) ?? new SortedSet<string>();
```

Comment:
Will `OptionSettingValueType.List` -> `SortedSet` potentially drop entries for list options that accept duplicate values?

Even if there aren't any examples today, wouldn't want to make this fragile since it'd be tough to remember to come back and update this.

Reply:
Your concerns are valid. However, for all option settings of type `OptionSettingValueType.List`, we are already processing them as a `SortedSet`

* [Exhibit A](https://github.com/aws/aws-dotnet-deploy/blob/66fb1e34bbea10bbd995596e1eec801b79ff9102/src/AWS.Deploy.CLI/Commands/DeployCommand.cs#L737)
* [Exhibit B](https://github.com/aws/aws-dotnet-deploy/blob/66fb1e34bbea10bbd995596e1eec801b79ff9102/src/AWS.Deploy.Common/Recipes/OptionSettingItem.ValueOverride.cs#L139)
* [Exhibit C](https://github.com/aws/aws-dotnet-deploy/blob/66fb1e34bbea10bbd995596e1eec801b79ff9102/src/AWS.Deploy.Orchestration/OptionSettingHandler.cs#L299)

Should those occurences be changed as well? If so, I'd prefer to to do that (along with the new instance) in a seperate PR.